data/helpers/conversations/upsertConversations.ts

```diff
@@ -77,10 +81,20 @@ const setupAndSaveConversations = async (
   });
 
   const conversationsToUpsert: Conversation[] = [];
-  conversations.forEach((conversation) => {
+  conversations.forEach(async (conversation) => {
```

Comment:
using `async` in a `forEach` will not make the `forEach` wait for the `await` 
also this is ou conversation `upsert` method that's supposed to save stuff in the db, I think it should be as fast as possible